### PR TITLE
DB 2103 more robust Banner Message

### PR DESCRIPTION
**Pull request by kimminahm**
- [x] Merged concurrently with front end https://github.com/fedspendingtransparency/data-act-broker-web-app/pull/674
- [x] Reviewed by backend

Alembic upgrade, updating backend table to be more robust. Adds more information to the response so front end can understand it
Allows for more customization, FABS vs DABS vs site wide announcements

### CODE REVIEW

**Comment by alburde1** [comment]
Also a note, another alembic migration went in so you're going to have to adjust yours.

File: dataactcore/models/jobModels.py [hunk 1]
```diff
@@ -236,3 +236,12 @@ class SubmissionWindow(Base):
     end_date = Column(Date)
     block_certification = Column(Boolean, default=False)
     message = Column(Text)
+    application_type = Column(Integer, ForeignKey("application_type.application_id", name="application"))
```
Comment (kimminahm) [inline dataactcore/models/jobModels.py]: changed

File: tests/unit/dataactbroker/test_fileRoutes.py [hunk 2]
```diff
@@ -80,14 +80,18 @@ def test_is_period(file_app, database):
     the number of submissions within the agency of the user that is logged in
     """
 
-    gtas = WindowFactory(start_date=datetime(2007, 1, 3), end_date=datetime(2010, 3, 5), block_certification=False)
+    application = ApplicationTypeFactory(application_id=1, application_name='fabs')
```
Comment (kimminahm) [inline tests/unit/dataactbroker/test_fileRoutes.py]: fixed

File: tests/unit/dataactbroker/test_fileRoutes.py [hunk 3]
```diff
@@ -80,14 +80,18 @@ def test_is_period(file_app, database):
...
+                         message='first', application_type=application.application_id)
     database.session.add(gtas)
 
     response = file_app.get("/v1/window/")
     response_json = json.loads(response.data.decode('UTF-8'))
     assert response_json['data'] is None
 
-    gtas = WindowFactory(start_date=datetime(2007, 1, 3), end_date=datetime(2010, 3, 5), block_certification=True)
+    gtas = WindowFactory(start_date=datetime(2007, 1, 3), end_date=datetime(2010, 3, 5), block_certification=True,
+                         message='first', application_type=application.application_id)
```
Comment (kimminahm) [inline tests/unit/dataactbroker/test_fileRoutes.py]: ok

File: tests/unit/dataactbroker/test_fileRoutes.py [hunk 4]
```diff
@@ -80,14 +80,20 @@ def test_is_period(file_app, database):
     the number of submissions within the agency of the user that is logged in
     """
 
-    gtas = WindowFactory(start_date=datetime(2007, 1, 3), end_date=datetime(2010, 3, 5), block_certification=False)
+    application = ApplicationTypeFactory(application_name='fabs')
+
```
Comment (alburde1) [inline tests/unit/dataactbroker/test_fileRoutes.py]: Do we need all this empty space?

File: tests/unit/dataactcore/factories/job.py [hunk 5]
```diff
@@ -110,6 +110,14 @@ class Meta:
...
 
+class ApplicationTypeFactory(factory.Factory):
+    class Meta:
+        model = jobModels.ApplicationType
+
+    application_type_id = fuzzy.FuzzyInteger(9999)
+    application_name = fuzzy.FuzzyText()
+
+
 class WindowFactory(factory.Factory):
```
Comment (alburde1) [inline tests/unit/dataactcore/factories/job.py]: Why is this called `WindowFactory` when the model is `SubmissionWindow`? It should be `SubmissionWindowFactory`

File: tests/unit/dataactcore/factories/job.py [hunk 6]
```diff
@@ -119,3 +127,5 @@ class Meta:
     end_date = fuzzy.FuzzyDate(date(2010, 1, 1))
     block_certification = False
     message = fuzzy.FuzzyText()
+    application_type = fuzzy.FuzzyInteger(9999)
```
Comment (alburde1) [inline tests/unit/dataactcore/factories/job.py]: When you add a SubFactory to a Factory, it's generally unnecessary to also add the ID you're using as the foreign key. Just pass it an object to use with the relationship and it'll figure out the rest

File: dataactcore/migrations/versions/e3477fa42630_submission_window_type_lookup.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
+"""Submission window type lookup
```
Comment (alburde1) [inline dataactcore/migrations/versions/e3477fa42630_submission_window_type_lookup.py]: I just realized this isn't really an accurate message. A more accurate message would be "Adding application_type table, updating submission_window table"

File: dataactcore/models/jobModels.py [hunk 8]
```diff
@@ -236,3 +236,13 @@ class SubmissionWindow(Base):
     end_date = Column(Date)
     block_certification = Column(Boolean, default=False)
     message = Column(Text)
+    application_type = Column(Integer, ForeignKey("application_type.application_type_id",
```
Comment (alburde1) [inline dataactcore/models/jobModels.py]: This should still be `application_type_id` to match the column it's pointing to

**Pull request merged**